tsdb/db.go

```diff
@@ -384,26 +384,36 @@ var ErrClosed = errors.New("db already closed")
 // Current implementation doesn't support concurrency so
 // all API calls should happen in the same go routine.
 type DBReadOnly struct {
-	logger  log.Logger
-	dir     string
-	closers []io.Closer
-	closed  chan struct{}
+	logger     log.Logger
+	dir        string
+	sandboxDir string
+	closers    []io.Closer
+	closed     chan struct{}
 }
 
 // OpenDBReadOnly opens DB in the given directory for read only operations.
-func OpenDBReadOnly(dir string, l log.Logger) (*DBReadOnly, error) {
+func OpenDBReadOnly(dir, sandboxDirRoot string, l log.Logger) (*DBReadOnly, error) {
 	if _, err := os.Stat(dir); err != nil {
 		return nil, fmt.Errorf("opening the db dir: %w", err)
 	}
 
+	if sandboxDirRoot == "" {
```

Comment:
The dir will be created for all `DBReadOnly`, even though it'll not be used (`Blocks()` will not use it e.g.).
Doing consistency and to simplify tests and to make tem future proof (avoid missing a case if a new one is added.)
wdyt? would that be confusing for users?

Reply:
Only when the read only tsdb is instantiated right? Does that happen even if we dont use promtool?

Reply:
All `OpenDBReadOnly` callers, we only use it in `promtool` on our end: https://github.com/search?q=repo%3Aprometheus%2Fprometheus+OpenDBReadOnly&type=code

Reply:
Then I think it is fine!